### PR TITLE
WT-4977 Migrate pull request clang analyzer to evergreen

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -1117,6 +1117,16 @@ tasks:
             set -o verbose
             ./test/evergreen/verify_wt_datafiles.sh 2>&1
 
+  - name: clang-analyzer
```
Comment (lukech) [inline test/evergreen.yml]: Depends on if WT-5176 is merged first, we need to tag this task as `pull_request`. 
Also, I'd prefer this new task being put next to other pull request tasks.

Comment (quchenhao) [reply test/evergreen.yml]: OK, we can merge that after WT-5176.

**Comment by quchenhao** [comment]
Sorry, accidentally triggered rereview. Ignore that.

**Comment by quchenhao** [comment]
What is Clang scan-build trend doing? I checked and I see it is producing a graph but I cannot interpret it.

**Comment by quchenhao** [comment]
Intentionally triggered one failed patch build. https://evergreen.mongodb.com/task/wiredtiger_ubuntu1804_clang_analyzer_patch_af2cb8f052184a94635c3bfc358620dd36df5828_5da4fa6f3066156c2416a923_19_10_14_22_46_44

**Comment by lukech** [comment]
Thanks for the failure test, it's good to confirm the new task is able to `fail`.

**Pull request merged**